pkg/app/app_wasm.go

```diff
@@ -139,7 +139,18 @@ func navigate(u *url.URL, updateHistory bool) error {
 
 	root, ok := routes[path]
 	if !ok {
-		root = NotFound
+		for i := range routesRe {
```

Comment:
Here it would be more readable if you use the for var assignment rather than the index :
```go
for _, r := range routesWithRegexp {
}
```

Reply:
I specifically didn't want to do this because I don't want to "copy" the handler object, because the handler object might have state, e..g., expects to be a singleton

Reply:
I specifically didn't want to do this because I don't want to "copy" the handler object, because the handler object might have state, e..g., expects to be a singleton

Reply:
here's more info on the use case...
the regexp url handler has a folder of encrypted web pages. it uses the url to figure out the document to load and decrypt, and then will insert the Raw() html into the page DOM. Before inserting the page, it rewrites all href links to make sure they conform with the regexp pattern so that it gets called to decrypt the next page...

Reply:
you were right. The struct has a pointer in it, so a singleton works with the for loop case. I wrote a unit test to confirm. Just updated app_wasm.go per your suggestion.